File: rehab_py/8kyu/8kyu_positive_sum.java
// class Positive {
//     public static int sum(int[] arr) {
//         int rn = 0;
//         for (int s : arr) {
//             if (s > 0) {
//                 rn += s;
//             }
//         }
//         return rn;
//     }
// }

class Positive {
    public static int sum(int[] arr) {
        int sum = 0;
        for (int num : arr)
            sum += num > 0 ? num : 0;
        return sum;
    }
}
// 兩者速度沒有差多少

File: rehab_py/8kyu/8kyu_positive_sum.py
def positive_sum(arr):
    return sum([n for n in arr if n >0])

def positive_sum(arr):
    return sum(x for x in arr if x > 0)

# ----------------------------------------------------------------------------
# ! list comprehension vs generator expression
# 兩者在小資料量上差異不代，但是list變大，就有差

# import timeit
# def positive_sum_1():
#     arr = list(range(-10_000, 10_000))
#     return sum([n for n in arr if n >0])

# def positive_sum_2():
#     arr = list(range(-10_000, 10_000))
#     return sum(x for x in arr if x > 0)

# ff = timeit.timeit(positive_sum_1, number=100)
# ss = timeit.timeit(positive_sum_2, number=100)

# print(ff,ss)
# 0.08756324099886115 0.09390460999929928